File: 2021/day2/main.ts
import * as fs from "fs";


function part1(): void {

    // read file
    const data: string = fs.readFileSync("day2/input.txt", "utf-8")
    const lines: string[] = data.split("\n")

    let x: number = 0
    let y: number = 0

    for (let line of lines) {
        if (line == "") {
            break
        }
        const vals: string[] = line.split(" ")
        const direction: string = vals[0]
        const step: number = parseInt(vals[1])
        // console.log(vals, direction)
        switch (direction) {
            case "forward":
                x += step
                break;
            case "down":
                y += step
                break;
            case "up":
                y -= step
                break;
        }
    }

    console.log("Part1: %d", (x*y))

}

function part2(): void {
    // read file
    const data: string = fs.readFileSync("day2/input.txt", "utf-8")
    const lines: string[] = data.split("\n")

    let x: number = 0
    let y: number = 0
    let z: number = 0

    for (let line of lines) {
        if (line == "") {
            break
        }
        const vals: string[] = line.split(" ")
        const direction: string = vals[0]
        const step: number = parseInt(vals[1])
        switch (direction) {
            case "forward":
                x += step
                z += step*y
                break;
            case "down":
                y += step
                break;
            case "up":
                y -= step
                break;
        }
    }

    console.log("Part2: %d", (x*z))

}

function main(): void {
    part1()
    part2()
}

main()

File: 2021/day2/main.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func part1() {
	// read file
	bytes, err := os.ReadFile("day2/input.txt")
	if err != nil {
		panic(err)
	}
	lines := strings.Split(string(bytes), "\n")

	x := 0
	y := 0

	for _, line := range lines {
		if line == "" {
			break
		}
		vals := strings.Split(line, " ")
		direction := vals[0]
		step, err := strconv.Atoi(vals[1])
		if err != nil {
			panic(err)
		}

		switch direction {
		case "forward":
			x += step
		case "down":
			y += step
		case "up":
			y -= step
		}

	}

	fmt.Printf("Part 1: %d\n", x * y)

}

func part2() {
	// read file
	bytes, err := os.ReadFile("day2/input.txt")
	if err != nil {
		panic(err)
	}
	lines := strings.Split(string(bytes), "\n")

	x := 0
	y := 0
	z := 0

	for _, line := range lines {
		if line == "" {
			break
		}
		vals := strings.Split(line, " ")
		direction := vals[0]
		step, err := strconv.Atoi(vals[1])
		if err != nil {
			panic(err)
		}

		switch direction {
		case "forward":
			x += step
			z += step *y
		case "down":
			y += step
		case "up":
			y -= step
		}

	}

	fmt.Printf("Part 2: %d\n", x * z)
}

func main() {
	part1()
	part2()
}

